src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -348,7 +348,12 @@
 
     List<Q> keysQueryMapperList =
         chunkedKeys.stream()
-            .map(_keys -> keysQueryMapper.apply(new HashSet<>(_keys)))
+            .map(
+                _keys ->
+                    keysQueryMapper
+                        .apply(new HashSet<>(_keys))
+                        .withLimit(_keys.size())
+                        .withFetchTotal(false))
```

Comment:
Do you see any advantage adding this 2 -> .withLimit and withFetchTotal ?
If yes, Can you add the same to cacheKeysToIds and cacheKeysToIdsUsingGraphQl methods as well? Or may be refactor to a method and use them. WDYT?

Reply:
I will check other parts, withFetchTotal is for performance. withLimit is necessary otherwise it would return only with the default limit and some of them would be missed to sync :) like in benchmarks

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/690/commits/3346ad0658a56af69876501ca48c9f606919f7aa